exporters/common/src/main/java/io/opentelemetry/exporter/internal/metrics/SemConvExporterMetrics.java

```diff
@@ -0,0 +1,209 @@
+ * any time.
+ */
+public class SemConvExporterMetrics implements ExporterMetrics {
+
+  private static final Clock CLOCK = Clock.getDefault();
+
+  private final Supplier<MeterProvider> meterProviderSupplier;
+  private final ComponentId.StandardExporterType.Signal signal;
+  private final ComponentId componentId;
+  private final Attributes additionalAttributes;
+
+  @Nullable private volatile LongUpDownCounter inflight = null;
+  @Nullable private volatile LongCounter exported = null;
+  @Nullable private volatile DoubleHistogram duration = null;
+  @Nullable private volatile Attributes allAttributes = null;
+
+  public SemConvExporterMetrics(
+      Supplier<MeterProvider> meterProviderSupplier,
+      ComponentId.StandardExporterType.Signal signal,
+      ComponentId componentId,
+      Attributes additionalAttributes) {
+    this.meterProviderSupplier = meterProviderSupplier;
+    this.componentId = componentId;
+    this.signal = signal;
+    this.additionalAttributes = additionalAttributes;
+  }
+
+  @Override
+  public ExporterMetrics.Recording startRecordingExport(int itemCount) {
+    return new Recording(itemCount);
+  }
+
+  private static String getNamespace(ComponentId.StandardExporterType.Signal signal) {
+    switch (signal) {
+      case SPAN:
+        return "otel.sdk.exporter.span";
+      case METRIC:
+        return "otel.sdk.exporter.metric_data_point";
+      case LOG:
+        return "otel.sdk.exporter.log";
+    }
+    throw new IllegalArgumentException("Unhandled signal: " + signal);
+  }
+
+  private static String getUnit(ComponentId.StandardExporterType.Signal signal) {
+    switch (signal) {
+      case SPAN:
+        return "span";
+      case METRIC:
+        return "data_point";
+      case LOG:
+        return "log_record";
+    }
+    throw new IllegalArgumentException("Unhandled signal: " + signal);
+  }
+
+  private Meter meter() {
+    MeterProvider meterProvider = meterProviderSupplier.get();
+    if (meterProvider == null) {
+      meterProvider = MeterProvider.noop();
+    }
+    return meterProvider.get("io.opentelemetry.exporters." + componentId.getTypeName());
+  }
+
+  private Attributes allAttributes() {
+    // attributes are initialized lazily to trigger lazy initialization of the componentId
+    Attributes allAttributes = this.allAttributes;
+    if (allAttributes == null) {
+      AttributesBuilder builder = Attributes.builder();
+      componentId.put(builder);
+      builder.putAll(additionalAttributes);
+      allAttributes = builder.build();
+      this.allAttributes = allAttributes;
+    }
+    return allAttributes;
+  }
+
+  private LongUpDownCounter inflight() {
+    LongUpDownCounter inflight = this.inflight;
+    if (inflight == null || isNoop(inflight)) {
+      String unit = getUnit(signal);
+      inflight =
+          meter()
+              .upDownCounterBuilder(getNamespace(signal) + ".inflight")
+              .setUnit("{" + unit + "}")
```

Comment:
#nit: its strange to have a method called `getUnit` which needs to be formatted with `{` `}` before calling `setUnit` with it. Let's embed wrapping with `{` `}` in the `getUnit` method.

Reply:
Don't know if you noticed, but the unit is also used in the metric description below. What would you prefer of the following options:
 * Keep the unit as is without `{}`
 * Add the surrounding `{}` to the unit and remove the braces for the description text
 * Add the surrounding `{}` to the unit and keep the braces even in the description text
 
I'm happy with any option